<a href="https://colab.research.google.com/github/mudesir/DS-Unit-1-Sprint-2-Statistics/blob/master/Mudesir_DS17_121_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

In [3]:
### From the raw data we see that the dataset doesn't have column headers and have some missed values lets 
# fix the header and the cell values assigned by "?" clean our data
import pandas as pd
column_headers = ['party', 'handicapped-infants', 'water-project-cost-sharing', 'adoption-of-the-budget-resolution', 
                    'physician-fee-freeze', 'el-salvador-aid', 'religious-groups-in-schools', 'anti-satellite-test-ban', 
                    'aid-to-nicaraguan-contras', 'mx-missile', 'immigration', 'synfuels-corporation-cutback', 'education-spending', 
                    'superfund-right-to-sue', 'crime', 'duty-free-exports', 'export-administration-act-south-africa']
df = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data', names=column_headers)
df.head()                                                                                                                          

party  ... export-administration-act-south-africa
0  republican  ...                                      y
1  republican  ...                                      ?
2    democrat  ...                                      n
3    democrat  ...                                      y
4    democrat  ...                                      y

[5 rows x 17 columns]

In [4]:
df.dtypes

party                                     object
handicapped-infants                       object
water-project-cost-sharing                object
adoption-of-the-budget-resolution         object
physician-fee-freeze                      object
el-salvador-aid                           object
religious-groups-in-schools               object
anti-satellite-test-ban                   object
aid-to-nicaraguan-contras                 object
mx-missile                                object
immigration                               object
synfuels-corporation-cutback              object
education-spending                        object
superfund-right-to-sue                    object
crime                                     object
duty-free-exports                         object
export-administration-act-south-africa    object
dtype: object

In [5]:
df.isnull().sum()

party                                     0
handicapped-infants                       0
water-project-cost-sharing                0
adoption-of-the-budget-resolution         0
physician-fee-freeze                      0
el-salvador-aid                           0
religious-groups-in-schools               0
anti-satellite-test-ban                   0
aid-to-nicaraguan-contras                 0
mx-missile                                0
immigration                               0
synfuels-corporation-cutback              0
education-spending                        0
superfund-right-to-sue                    0
crime                                     0
duty-free-exports                         0
export-administration-act-south-africa    0
dtype: int64

In [6]:
df.shape

(435, 17)

In [7]:
import numpy as np
# Replace 'y'(yes) to '1' , 'n' to '0', and '?' ton NaN
df = df.replace({'y': 1, 'n': 0, '?': np.NaN})
print(df.shape)
df.head()

(435, 17)


party  ...  export-administration-act-south-africa
0  republican  ...                                     1.0
1  republican  ...                                     NaN
2    democrat  ...                                     0.0
3    democrat  ...                                     1.0
4    democrat  ...                                     1.0

[5 rows x 17 columns]

In [8]:
df.isnull().sum()

party                                       0
handicapped-infants                        12
water-project-cost-sharing                 48
adoption-of-the-budget-resolution          11
physician-fee-freeze                       11
el-salvador-aid                            15
religious-groups-in-schools                11
anti-satellite-test-ban                    14
aid-to-nicaraguan-contras                  15
mx-missile                                 22
immigration                                 7
synfuels-corporation-cutback               21
education-spending                         31
superfund-right-to-sue                     25
crime                                      17
duty-free-exports                          28
export-administration-act-south-africa    104
dtype: int64

In [9]:
df['party'].value_counts()

democrat      267
republican    168
Name: party, dtype: int64

In [0]:
# Create dataframe by party
dem = df[df['party']=='democrat']
rep = df[df['party']=='republican']

In [11]:
dem.head()

party  ...  export-administration-act-south-africa
2  democrat  ...                                     0.0
3  democrat  ...                                     1.0
4  democrat  ...                                     1.0
5  democrat  ...                                     1.0
6  democrat  ...                                     1.0

[5 rows x 17 columns]

In [12]:
rep.head()


party  ...  export-administration-act-south-africa
0   republican  ...                                     1.0
1   republican  ...                                     NaN
7   republican  ...                                     1.0
8   republican  ...                                     1.0
10  republican  ...                                     0.0

[5 rows x 17 columns]

## 2 sample t-test on Bills vote democrats support more than republicans 
1. Null Hypothhesis: democrats support the same proportion as republican on Bills
$H_0: \bar{x}_d = \bar{x}_r$
2. Alternative Hypothesis: democrats support are not equal with republican on bill
$H_0: \bar{x}_d  !=  \bar{x}_r$
3. Confidence level: 99%

The Bills are: ***'handicapped-infants', 'water-project-cost-sharing', 'adoption-of-the-budget-resolution', 'physician-fee-freeze', 'el-salvador-aid', 'religious-groups-in-schools', 'anti-satellite-test-ban', 'aid-to-nicaraguan-contras', 'mx-missile', 'immigration',           'synfuels-corporation-cutback', 'education-spending', 'superfund-right-to-sue', 'crime', 'duty-free-exports', 'export-administration-act-south-africa'***

In [13]:
#handicapped-infants
y1 = dem[dem['handicapped-infants']==1]
y2 = rep[rep['handicapped-infants']==1]
y=y1['handicapped-infants'].value_counts() + y2['handicapped-infants'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['handicapped-infants']==0]
n2 = rep[rep['handicapped-infants']==0]
n=n1['handicapped-infants'].value_counts() + n2['handicapped-infants'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
from scipy import stats
stats.ttest_ind(dem['handicapped-infants'], rep['handicapped-infants'],nan_policy='omit')

yse vote:
 1.0    187
Name: handicapped-infants, dtype: int64
no vote:
 0.0    236
Name: handicapped-infants, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=9.205264294809222, pvalue=1.613440327937243e-18)

**Conclusion**: Given a t-statistic of 9.2, and p-value of 0, we **reject** the null hypothesis that democrats suport for **handicapped-infants** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statiscs the Democrat support the bill **more than** the republicans.

In [14]:
# water-project-cost-sharing
y1 = dem[dem['water-project-cost-sharing']==1]
y2 = rep[rep['water-project-cost-sharing']==1]
y=y1['water-project-cost-sharing'].value_counts() + y2['water-project-cost-sharing'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['water-project-cost-sharing']==0]
n2 = rep[rep['water-project-cost-sharing']==0]
n=n1['water-project-cost-sharing'].value_counts() + n2['water-project-cost-sharing'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
  # 2t-test
stats.ttest_ind(dem['water-project-cost-sharing'], rep['water-project-cost-sharing'],nan_policy='omit')

yse vote:
 1.0    195
Name: water-project-cost-sharing, dtype: int64
no vote:
 0.0    192
Name: water-project-cost-sharing, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=-0.08896538137868286, pvalue=0.9291556823993485)

**Conclusion**: Given a t-statistic of 0.089, and p-value of 0.929, we **fail to reject** the null hypothesis that democrats suport for **water-project-cost-sharing** bill is equal to repabilican. And also the  Republicans support the bill **more than** the Democrat.

In [15]:
# adoption-of-the-budget-resolution
y1 = dem[dem['adoption-of-the-budget-resolution']==1]
y2 = rep[rep['adoption-of-the-budget-resolution']==1]
y=y1['adoption-of-the-budget-resolution'].value_counts() + y2['adoption-of-the-budget-resolution'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['adoption-of-the-budget-resolution']==0]
n2 = rep[rep['adoption-of-the-budget-resolution']==0]
n=n1['adoption-of-the-budget-resolution'].value_counts() + n2['adoption-of-the-budget-resolution'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
  # 2t-test
stats.ttest_ind(dem['adoption-of-the-budget-resolution'], rep['adoption-of-the-budget-resolution'], nan_policy='omit')

yse vote:
 1.0    253
Name: adoption-of-the-budget-resolution, dtype: int64
no vote:
 0.0    171
Name: adoption-of-the-budget-resolution, dtype: int64
The bill passed by majority vote


Ttest_indResult(statistic=23.21277691701378, pvalue=2.0703402795404463e-77)

**Conclusion**: Given a t-statistic of 23.213, and p-value of 0, we **reject** the null hypothesis that democrats suport for **adoption-of-the-budget-resolution** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statiscs the Democrat support the bill **more than** the republicans.

In [17]:
# physician-fee-freeze
y1 = dem[dem['physician-fee-freeze']==1]
y2 = rep[rep['physician-fee-freeze']==1]
y=y1['physician-fee-freeze'].value_counts() + y2['physician-fee-freeze'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['physician-fee-freeze']==0]
n2 = rep[rep['physician-fee-freeze']==0]
n=n1['physician-fee-freeze'].value_counts() + n2['physician-fee-freeze'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['physician-fee-freeze'], rep['physician-fee-freeze'],nan_policy='omit')

yse vote:
 1.0    177
Name: physician-fee-freeze, dtype: int64
no vote:
 0.0    247
Name: physician-fee-freeze, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=-49.36708157301406, pvalue=1.994262314074344e-177)

Conclusion: Given a t-statistic of -49.37, and p-value of 0, we **reject** the null hypothesis that democrats suport for **physician fee freeze** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statiscs the Republicans support the bill **more than** the Democrat.

In [18]:
# el-salvador-aid
y1 = dem[dem['el-salvador-aid']==1]
y2 = rep[rep['el-salvador-aid']==1]
y=y1['el-salvador-aid'].value_counts() + y2['el-salvador-aid'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['el-salvador-aid']==0]
n2 = rep[rep['el-salvador-aid']==0]
n=n1['el-salvador-aid'].value_counts() + n2['el-salvador-aid'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['el-salvador-aid'], rep['el-salvador-aid'],nan_policy='omit')

yse vote:
 1.0    212
Name: el-salvador-aid, dtype: int64
no vote:
 0.0    208
Name: el-salvador-aid, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=-21.13669261173219, pvalue=5.600520111729011e-68)

**Conclusion**: Given a t-statistic of -21.137, and p-value of 0, we **reject** the null hypothesis that democrats suport for **el-salvador-aid** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **less than** the republicans.

In [21]:
# religious-groups-in-schools
y1 = dem[dem['religious-groups-in-schools']==1]
y2 = rep[rep['religious-groups-in-schools']==1]
y=y1['religious-groups-in-schools'].value_counts() + y2['religious-groups-in-schools'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['religious-groups-in-schools']==0]
n2 = rep[rep['religious-groups-in-schools']==0]
n=n1['religious-groups-in-schools'].value_counts() + n2['religious-groups-in-schools'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['religious-groups-in-schools'], rep['religious-groups-in-schools'],nan_policy='omit')

yse vote:
 1.0    272
Name: religious-groups-in-schools, dtype: int64
no vote:
 0.0    152
Name: religious-groups-in-schools, dtype: int64
The bill passed by majority vote


Ttest_indResult(statistic=-9.737575825219457, pvalue=2.3936722520597287e-20)

**Conclusion**: Given a t-statistic of -9.738, and p-value of 0, we **reject** the null hypothesis that democrats suport for **religious-groups-in-schools** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **less than** the republicans.

In [20]:
# anti-satellite-test-ban
y1 = dem[dem['anti-satellite-test-ban']==1]
y2 = rep[rep['anti-satellite-test-ban']==1]
y=y1['anti-satellite-test-ban'].value_counts() + y2['anti-satellite-test-ban'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['anti-satellite-test-ban']==0]
n2 = rep[rep['anti-satellite-test-ban']==0]
n=n1['anti-satellite-test-ban'].value_counts() + n2['anti-satellite-test-ban'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['anti-satellite-test-ban'], rep['anti-satellite-test-ban'],nan_policy='omit')

yse vote:
 1.0    239
Name: anti-satellite-test-ban, dtype: int64
no vote:
 0.0    182
Name: anti-satellite-test-ban, dtype: int64
The bill passed by majority vote


Ttest_indResult(statistic=12.526187929077842, pvalue=8.521033017443867e-31)

**Conclusion**: Given a t-statistic of 12.526, and p-value of 0, we **reject** the null hypothesis that democrats suport for **anti-satellite-test** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **more than** the republicans.

In [22]:
# aid-to-nicaraguan-contras
y1 = dem[dem['aid-to-nicaraguan-contras']==1]
y2 = rep[rep['aid-to-nicaraguan-contras']==1]
y=y1['aid-to-nicaraguan-contras'].value_counts() + y2['aid-to-nicaraguan-contras'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['aid-to-nicaraguan-contras']==0]
n2 = rep[rep['aid-to-nicaraguan-contras']==0]
n=n1['aid-to-nicaraguan-contras'].value_counts() + n2['aid-to-nicaraguan-contras'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['aid-to-nicaraguan-contras'], rep['aid-to-nicaraguan-contras'],nan_policy='omit')

yse vote:
 1.0    242
Name: aid-to-nicaraguan-contras, dtype: int64
no vote:
 0.0    178
Name: aid-to-nicaraguan-contras, dtype: int64
The bill passed by majority vote


Ttest_indResult(statistic=18.052093200819733, pvalue=2.82471841372357e-54)

**Conclusion**: Given a t-statistic of 18.052, and p-value of 0, we **reject** the null hypothesis that democrats suport for **aid-to-nicaraguan-cotras** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **more than** the republicans.

In [23]:
# mx-missile
y1 = dem[dem['mx-missile']==1]
y2 = rep[rep['mx-missile']==1]
y=y1['mx-missile'].value_counts() + y2['mx-missile'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['mx-missile']==0]
n2 = rep[rep['mx-missile']==0]
n=n1['mx-missile'].value_counts() + n2['mx-missile'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['mx-missile'], rep['mx-missile'],nan_policy='omit')

yse vote:
 1.0    207
Name: mx-missile, dtype: int64
no vote:
 0.0    206
Name: mx-missile, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=16.437503268542994, pvalue=5.03079265310811e-47)

**Conclusion**: Given a t-statistic of 16.438, and p-value of 0, we **reject** the null hypothesis that democrats suport for **mx-missile** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **more than** the republicans.

In [24]:
# immigration
y1 = dem[dem['immigration']==1]
y2 = rep[rep['immigration']==1]
y=y1['immigration'].value_counts() + y2['immigration'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['immigration']==0]
n2 = rep[rep['immigration']==0]
n=n1['immigration'].value_counts() + n2['immigration'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['immigration'], rep['immigration'],nan_policy='omit')

yse vote:
 1.0    216
Name: immigration, dtype: int64
no vote:
 0.0    212
Name: immigration, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=-1.7359117329695164, pvalue=0.08330248490425066)

**Conclusion**: Given a t-statistic of -1.736, and p-value of 0.083, we **fail to reject** the null hypothesis that democrats suport for **immigration** bill is equal to repabilican. And also from the sign of t-statistic the Democrat support the bill **less than** the republicans.

In [25]:
# synfuels-corporation-cutback
y1 = dem[dem['synfuels-corporation-cutback']==1]
y2 = rep[rep['synfuels-corporation-cutback']==1]
y=y1['synfuels-corporation-cutback'].value_counts() + y2['synfuels-corporation-cutback'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['synfuels-corporation-cutback']==0]
n2 = rep[rep['synfuels-corporation-cutback']==0]
n=n1['synfuels-corporation-cutback'].value_counts() + n2['synfuels-corporation-cutback'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['synfuels-corporation-cutback'], rep['synfuels-corporation-cutback'],nan_policy='omit')

yse vote:
 1.0    150
Name: synfuels-corporation-cutback, dtype: int64
no vote:
 0.0    264
Name: synfuels-corporation-cutback, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=8.293603989407588, pvalue=1.5759322301054064e-15)

**Conclusion**: Given a t-statistic of 8.294, and p-value of 0, we **reject** the null hypothesis that democrats suport for **synfuels-corporation-cutback** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **more than** the republicans.

In [26]:
# education-spending
y1 = dem[dem['education-spending']==1]
y2 = rep[rep['education-spending']==1]
y=y1['education-spending'].value_counts() + y2['education-spending'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['education-spending']==0]
n2 = rep[rep['education-spending']==0]
n=n1['education-spending'].value_counts() + n2['education-spending'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['education-spending'], rep['education-spending'],nan_policy='omit')

yse vote:
 1.0    171
Name: education-spending, dtype: int64
no vote:
 0.0    233
Name: education-spending, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=-20.500685724563073, pvalue=1.8834203990450192e-64)

**Conclusion**: Given a t-statistic of -20.501, and p-value of 0, we **reject** the null hypothesis that democrats suport for **education-spending** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **less than** the republicans.

In [27]:
# superfund-right-to-sue
y1 = dem[dem['superfund-right-to-sue']==1]
y2 = rep[rep['superfund-right-to-sue']==1]
y=y1['superfund-right-to-sue'].value_counts() + y2['superfund-right-to-sue'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['superfund-right-to-sue']==0]
n2 = rep[rep['superfund-right-to-sue']==0]
n=n1['superfund-right-to-sue'].value_counts() + n2['superfund-right-to-sue'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['superfund-right-to-sue'], rep['superfund-right-to-sue'],nan_policy='omit')

yse vote:
 1.0    209
Name: superfund-right-to-sue, dtype: int64
no vote:
 0.0    201
Name: superfund-right-to-sue, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=-13.51064251060933, pvalue=1.2278581709672758e-34)

**Conclusion**: Given a t-statistic of -13.511, and p-value of 0, we **reject** the null hypothesis that democrats suport for **superfund-right-to-sue** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **less than** the republicans.

In [28]:
# crime
y1 = dem[dem['crime']==1]
y2 = rep[rep['crime']==1]
y=y1['crime'].value_counts() + y2['crime'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['crime']==0]
n2 = rep[rep['crime']==0]
n=n1['crime'].value_counts() + n2['crime'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['crime'], rep['crime'],nan_policy='omit')

yse vote:
 1.0    248
Name: crime, dtype: int64
no vote:
 0.0    170
Name: crime, dtype: int64
The bill passed by majority vote


Ttest_indResult(statistic=-16.342085656197696, pvalue=9.952342705606092e-47)

**Conclusion**: Given a t-statistic of -16.342, and p-value of 0, we **reject** the null hypothesis that democrats suport for **crime** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **less than** the republicans.

In [29]:
# duty-free-exports
y1 = dem[dem['duty-free-exports']==1]
y2 = rep[rep['duty-free-exports']==1]
y=y1['duty-free-exports'].value_counts() + y2['duty-free-exports'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['duty-free-exports']==0]
n2 = rep[rep['duty-free-exports']==0]
n=n1['duty-free-exports'].value_counts() + n2['duty-free-exports'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['duty-free-exports'], rep['duty-free-exports'],nan_policy='omit')

yse vote:
 1.0    174
Name: duty-free-exports, dtype: int64
no vote:
 0.0    233
Name: duty-free-exports, dtype: int64
The bill fail to pass


Ttest_indResult(statistic=12.853146132542978, pvalue=5.997697174347365e-32)

**Conclusion**: Given a t-statistic of 12.853, and p-value of 0, we **reject** the null hypothesis that democrats suport for **duty-free-exports** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **more than** the republicans.

In [30]:
# export-administration-act-south-africa
y1 = dem[dem['export-administration-act-south-africa']==1]
y2 = rep[rep['export-administration-act-south-africa']==1]
y=y1['export-administration-act-south-africa'].value_counts() + y2['export-administration-act-south-africa'].value_counts()
print('yse vote:\n',y)
n1 = dem[dem['export-administration-act-south-africa']==0]
n2 = rep[rep['export-administration-act-south-africa']==0]
n=n1['export-administration-act-south-africa'].value_counts() + n2['export-administration-act-south-africa'].value_counts()
print('no vote:\n', n)
if (y.sum()>218):
  print('The bill passed by majority vote')
else:
  print('The bill fail to pass')
stats.ttest_ind(dem['export-administration-act-south-africa'], rep['export-administration-act-south-africa'],nan_policy='omit')

yse vote:
 1.0    269
Name: export-administration-act-south-africa, dtype: int64
no vote:
 0.0    62
Name: export-administration-act-south-africa, dtype: int64
The bill passed by majority vote


Ttest_indResult(statistic=6.849454815841208, pvalue=3.652674361672226e-11)

**Conclusion**: Given a t-statistic of 6.849, and p-value of 0, we **reject** the null hypothesis that democrats suport for **export-adminstration-act-south-africa** bill is equal to repabilican. This suggests the alternative that the democrats do not support as republican on this bill. And also from the sign of t-statistic the Democrat support the bill **more than** the republicans.

## Bills Democrats support more than Republicans with p < 0.01


1.   handicapped-infants
2.   adoption-of-the-budget-resolution
3.   anti-satellite-test-ban
4.   aid-to-nicaraguan-contras
5.   mx-missile
6.   synfuels-corporation-cutback
7.   duty-free-exports
8.   export-administration-act-south-africa

## Bills Republicans support more than Democrats with p < 0.01


1.   water-project-cost-sharing
2.   physician-fee-freeze
3.   elsador-aid
4.   religious-groups-in-schools
5.   immigration
6.   education-spending
7.   superfund-right-to-sue
8.   crime

## Republicans and Democrats has p > 0.1


1.   water-project-cost-sharing






## Stretch Goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Work on Performing a T-test without using Scipy in order to get "under the hood" and learn more thoroughly about this topic.
### Start with a 1-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://blog.minitab.com/hs-fs/hubfs/Imported_Blog_Media/701f9c0efa98a38fb397f3c3ec459b66.png?width=247&height=172&name=701f9c0efa98a38fb397f3c3ec459b66.png) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then try a 2-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://lh3.googleusercontent.com/proxy/rJJ5ZOL9ZDvKOOeBihXoZDgfk7uv1YsRzSQ1Tc10RX-r2HrRpRLVqlE9CWX23csYQXcTniFwlBg3H-qR8MKJPBGnjwndqlhDX3JxoDE5Yg) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then check your Answers using Scipy!

## Establishing the one sample t-test I take the democrat vote for immigration mu=0.5 they vote 50 50 percent for yes and no:

1) Null Hypothesis: This is a fair vote. 

$\mu == \bar{x}$

My belief (my concept of the population) is that this is a fair vote (p=.5) => $\mu$

What the sample is telling me: $\bar{x}$

2) Alternative Hypothesis (Opposite of the null)

$\mu \neq \bar{x}$

The sample is different enough from my concept of the population, that it's unlikely that my sample came from a fair vote. 

3) Confidence Level: 99%




In [16]:
# The mean and standard deviation
dem_immig = dem['immigration']
dem_immig.dropna(inplace=True)
dem_immig.mean()

/usr/local/lib/python3.6/dist-packages/pandas/core/series.py:4494: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(result)


0.4714828897338403

In [17]:
dem_immig.std()

0.500137849978314

In [18]:
# xbar=.47 and sd=.5
# calculate t value
t = ((dem_immig.mean() - 0.5))/((dem_immig.std())/(len(dem_immig)**.5))
print(len(dem_immig))
print(t)

263
-0.9246846884388147


In [0]:
# degree of freedom = 263-1=262
# P-value = 1-.99=0.01
# t-calculated =-0.9316899756066176
# the P-value we get from t-table = 0.356
# conclussion: Given a t-statistic of -0.93, and p-value of 0.35, we fail to reject 
# the null hypothesis that democrats vote for immigration bill is fair 50% vote yes and 50% vote no.

In [19]:
# check with scipy
dem_immig = dem['immigration']
print(dem_immig.mean())
stats.ttest_1samp(dem_immig, .01, nan_policy='omit')

0.4714828897338403


Ttest_1sampResult(statistic=14.963864084775844, pvalue=5.206147515096537e-37)

## 2 sample t test I take the dem vote for immigration and crime x1 and x2 respectively for sample average 


1) Null Hypothesis: This is a fair vote. 

$\bar{x1} == \bar{x2}$



2) Alternative Hypothesis (Opposite of the null)

$\bar{x1} \neq \bar{x2}$

 

3) Confidence Level: 99%

In [20]:
xbar1 = dem_immig.mean()
s1 = dem_immig.std()
n1 = len(dem_immig)
print(xbar1,'\n')
print(s1,'\n')
print(n1)

0.4714828897338403 

0.500137849978314 

263


In [21]:
dem_crime = dem['crime']
dem_crime.dropna(inplace=True)
xbar2=dem_crime.mean()
s2 = dem_crime.std()
n2 = len(dem_crime)
print(xbar2,'\n')
print(s2,'\n')
print(n2)


0.35019455252918286 

0.4779615336263577 

257


/usr/local/lib/python3.6/dist-packages/pandas/core/series.py:4494: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._update_inplace(result)


In [22]:
# T=(xbar1−x2)/(sp*((1/n1)+(1/n2)**.5)) where sp=((n1-1)*s1**2 + (n2-1)*s2**2)**.5
sp = ((n1-1)*s1**2 + (n2-1)*s2**2)**.5
T = (xbar1-xbar2)/(sp*((1/n1) + (1/n2)**.5))
print(T)

0.16456789797611834


In [0]:
# degree of freedom = v = n1+n2-2
# p-value = (1-.99)/2 =0.005
# T = 0.16
# from the table v=500 and T=0.16 the p-value is 0.127

In [23]:
# using scipy
stats.ttest_ind(dem_immig, dem_crime, nan_policy='omit')

Ttest_indResult(statistic=2.8260722003326384, pvalue=0.004894258309215697)